phpunit/fonts-api/gutenbergRegisterFontsFromThemeJson-test.php

```diff
@@ -0,0 +1,360 @@
+		switch_theme( static::FONTS_THEME );
+
+		// Register Lato font.
+		wp_register_fonts(
+			array(
+				'Lato' => array(
+					array(
+						'font-family' => 'Lato',
+						'font-style'  => 'normal',
+						'font-weight' => '400',
+						'src'         => 'https://example.com/tests/assets/fonts/lato/Lato-Regular.woff2',
+					),
+					array(
+						'font-family' => 'Lato',
+						'font-style'  => 'italic',
+						'font-weight' => '400',
+						'src'         => 'https://example.com/tests/assets/fonts/lato/Lato-Regular-Italic.woff2',
+					),
+				),
+			)
+		);
+
+		// Pre-check to ensure no fonts are enqueued.
+		$this->assertEmpty( wp_fonts()->get_enqueued(), 'No fonts should be enqueued before running gutenberg_register_fonts_from_theme_json()' );
+
+		/*
+		 * When this function runs, it invokes WP_Theme_JSON_Resolver_Gutenberg::get_merged_data(),
+		 * which will include the Lato fonts with a 'origin' property set in each variation.
+		 */
+		gutenberg_register_fonts_from_theme_json();
+
+		$actual_enqueued_fonts = wp_fonts()->get_enqueued();
+
+		$this->assertNotContains( 'lato', $actual_enqueued_fonts, 'Lato font-family should not be enqueued' );
+		$this->assertSameSets( static::FONT_FAMILIIES[ static::FONTS_THEME ], $actual_enqueued_fonts, 'Only the theme font families should be enqueued' );
+	}
+
+	public function test_should_skip_when_font_face_not_defined() {
+		switch_theme( static::FONTS_THEME );
+		$expected_font_family = 'source-serif-pro';
+
+		/**
+		 * Callback that removes the 'fontFace' of the expected font family from the theme's theme.json data.
+		 * This callback is invoked at the start of gutenberg_register_fonts_from_theme_json() before processing
+		 * within that function. How? It's in the call stack of WP_Theme_JSON_Resolver_Gutenberg::get_merged_data().
+		 *
+		 * @param WP_Theme_JSON_Data_Gutenberg| WP_Theme_JSON_Data $theme_json_data Instance of the Data object.
+		 * @return WP_Theme_JSON_Data_Gutenberg| WP_Theme_JSON_Data Modified instance.
+		 * @throws ReflectionException
+		 */
+		$remove_expected_font_family = static function( $theme_json_data ) use ( $expected_font_family ) {
+			// Need to get the underlying data array which is in WP_Theme_JSON_Gutenberg | WP_Theme_JSON object.
+			$property = new ReflectionProperty( $theme_json_data, 'theme_json' );
+			$property->setAccessible( true );
+			$theme_json_object = $property->getValue( $theme_json_data );
+
+			$property = new ReflectionProperty( $theme_json_object, 'theme_json' );
+			$property->setAccessible( true );
+			$data = $property->getValue( $theme_json_object );
+
+			// Loop through the fonts to find the expected font-family to modify.
+			foreach ( $data['settings']['typography']['fontFamilies']['theme'] as $index => $definitions ) {
+				if ( $expected_font_family !== $definitions['slug'] ) {
+					continue;
+				}
+
+				// Remove the 'fontFace' element, which removes the font's varations.
```

Comment:
```suggestion
				// Remove the 'fontFace' element, which removes the font's variations.
```

Reply:
Suggestion committed in https://github.com/WordPress/gutenberg/pull/49877/commits/aeedd6d8091f41d1f53d46aacc241b52fe636608. Thank you!